sotodlib/core/flagman.py

```diff
@@ -0,0 +1,163 @@
+            else:
+                raise ValueError("FlagManager only takes data aligned with"
+                                     " dets and/or samps. Data of shape {}"
+                                     " is the wrong shape".format(s))
+        
+        if len(axis_map)==1 and axis_map[0][1]=='dets':
+            ### Change detector flags to RangesMatrix in the backend
+            x = Ranges(self.samps.count)
+            data = RangesMatrix([Ranges.ones_like(x) if Y 
+                                 else Ranges.zeros_like(x) for Y in data])
+            axis_map = [(0,'dets'),(1,'samps')]
+        super().wrap(name, data, axis_map, *kwargs)
+                    
+    def buffer(self, n_buffer, flags=None):
+        """Buffer all the samps cuts by n_buffer
+        Like with Ranges / Ranges Matrix, buffer changes everything in place        
+        
+        Arguments:
+        
+            n_buffer: number of samples to buffer the samps cuts
+            flags: List of flags to buffer. Uses their names
+        """
+        if flags is None:
+            flags = self._fields
+        
+        for f in flags:
+            self[f].buffer(n_buffer)
+        
+    def buffered(self, n_buffer, flags=None):
+        """Return new FlagManager that has all the samps cuts buffered by n_buffer
+        Like with Ranges / Ranges Matrix, buffered returns new object
+        
+        Arguments:
+        
+            n_buffer: number of samples to buffer the samps cuts
+            flags: List of flags to buffer. Uses their names
+        """
+        new = self.copy()
+        new.buffer(n_buffer, flags)
+        return new
+        
+    def collapse(self, flags=None, method='union'):
+        """Collapse flags in the FlagManager together. If flags is None
+        it returns all flags
+        
+        Arguments:
+        
+            flags: List of flags to collapse together. Uses their names.
+            method: How to collapse the data. Accepts 'union','intersect',
+                        or function.
+        """
+        if flags is None:
+            flags = self._fields
+
+        to_reduce = [self._fields[f] for f in flags]
+        if len(flags)==0:
+            raise ValueError('Found zero flags to combine')
+            
+        out = RangesMatrix([Ranges(self.samps.count) for det in self.dets.vals])
+        
+        ## need to add out to prevent flag ordering from causing errors
+        ### (Ranges can't add to RangeMatrix, only other way around)
+        to_reduce[0] = out+to_reduce[0]
+        
+        if method == 'union': 
+            op = lambda x, y: x+y
+        elif method == 'intersect':
+            op = lambda x, y: x*y
+        else:
+            op = method
+        out = reduce(op, to_reduce)
+        return out
+    
+    def collapse_all(self, method='union'):
+        """same as collapse but for all fields, to be more verbose"""
+        return self.collapse(new_flag, list(self._fields.keys()), method)
```

Comment:
This is typo... new_flag not defined.  Also it seems like this is intended to behave exactly like self.collapse() so I'm not sure you need it.